### PR TITLE
This exposes a simple run prompt method

### PR DESCRIPTION
**Pull request by jaresty**
- some of my custom extensions were failing because there was no longer an API for running a prompt and getting a response in talon script

### CODE REVIEW

File: GPT/gpt.py [hunk 1]
```diff
@@ -133,8 +133,16 @@ def gpt_select_last():
         for _ in lines[0]:
             actions.edit.extend_left()
 
-    def gpt_apply_prompt(mode: str, prompt: str, source: str, destination: str):
+    def gpt_apply_prompt(
+        mode: str, prompt: str, source: str, destination: str
+    ) -> dict[str, str]:
         """Apply an arbitrary prompt to arbitrary text"""
+        response = actions.user.gpt_run_prompt(mode, prompt, source)
+        actions.user.gpt_insert_response(format_message(response), destination, mode)
```
Comment (jaresty) [inline GPT/gpt.py:141]: notably we are unwrapping and rewrapping the message here. I went down the path of trying to skip doing that and it tended up being a much bigger change, so I think this is better for now.

**Comment by jaresty** [comment]
I also addressed #128

**Review by C-Loftus — APPROVED** [review]
Looks good to me. Thank you. My understanding of this PR,  correct me if I am wrong, is that we need to return the value into talon so then a user can run an arbitrary function of it if they would like.

I think a raw modifier may be interesting at some point, but not needed at the moment.

Minor merge conflict but feel free to merge once you are happy.

File: GPT/gpt.py (hunk 1 above)
Comment (C-Loftus) [reply GPT/gpt.py:141]: Fine with me. My plan is to eventually have a GPTMessage class that makes this a bit easier to reason about.

**Comment by jaresty** [comment]
That's exactly right-some of my custom scripts and I'm sure others as well were relying on the return value of the old function. This is a new api but at least it still works for that!

**Pull request merged**